Formula/jemalloc.rb

```diff
@@ -13,8 +12,26 @@ class Jemalloc < Formula
     sha256 "49a8f071338b3ec42cd89280681c338ad2e1d242a389e65f63f64e60257c5733" => :yosemite
   end
 
+  head do
+    url "https://github.com/jemalloc/jemalloc.git"
+    depends_on "autoconf" => :build
+    depends_on "docbook-xsl" => :build
+  end
+
   def install
-    system "./configure", "--disable-debug", "--prefix=#{prefix}", "--with-jemalloc-prefix="
+    args = %W[
+      --disable-debug
+      --prefix=#{prefix}
+      --with-jemalloc-prefix=
+    ]
+
+    if build.head?
+      args << "--with-xslroot=#{Formula["docbook-xsl"].opt_prefix}/docbook-xsl"
```

Comment:
does ` ENV["XML_CATALOG_FILES"] = etc/"xml/catalog"` not work?

Reply:
That is really bizarre. I'd like to know why it's ignoring the XML_CATALOG_FILES environment variable before we merge this.

Reply:
Some formulae use the same method:
- https://github.com/Homebrew/homebrew-core/blob/master/Formula/mkvtoolnix.rb#L60
- https://github.com/Homebrew/homebrew-core/blob/master/Formula/opensc.rb#L34
- https://github.com/Homebrew/homebrew-core/blob/master/Formula/postgis.rb#L69

Reply:
And 38 with XML_CATALOG_FILES …

Reply:
Log from failure build of jemalloc:
```
/usr/bin/xsltproc -o doc/jemalloc.html doc/html.xsl doc/jemalloc.xml
warning: failed to load external entity "/html/docbook.xsl"
compilation error: file doc/html.xsl line 2 element import
xsl:import : unable to load /html/docbook.xsl
make: *** [doc/jemalloc.html] Error 5
```
I hope this is somewhat helpful.

Reply:
BTW it's you who add `docbook-xsl` things into `mkvtoolnix` in 8e73a27.
Do you find any common point with `jemalloc` and `mkvtoolnix`?

Reply:
It looks like it's because jemalloc is using

```
href="/html/docbook.xsl"
```
instead of
```
href="http://docbook.sourceforge.net/release/xsl/current/html/docbook.xsl"
```
and we're only properly redirecting the latter to look in /usr/local/Cellar/docbook-xsl/1.79.1/docbook-xsl
```

So the question is whether we can tweak the docbook-xsl catalog to handle the `href="/html/docbook.xsl"` case. It's also not clear why jemalloc doesn't just use the full URIs.

Reply:
Yeah, so it works if I add
```
  <rewriteURI uriStartString="/" rewritePrefix="./"/>
  <rewriteSystem systemIdStartString="/" rewritePrefix="./"/>
```
to
```
/usr/local/Cellar/docbook-xsl/1.79.1/docbook-xsl/catalog.xml
```
We should check what Debian, etc. are doing here.

Reply:
So basically `jemalloc` is shooting itself in the foot by not using the full URI. If the concern is that they don't want to actually accidentally hit the Internet to access SourceForge, that's why xsltproc has the `--nonet` option.